test/DualTest.jl

```diff
@@ -15,6 +15,8 @@ samerng() = MersenneTwister(1)
 # exponent by one
 intrand(T) = T == Int ? rand(2:10) : rand(T)
 
+dualapprox(A, B) = value(A) ≈ value(B) && partials(A) ≈ partials(B)
```

Comment:
The (outdated-ly named) `test_approx_diffnums` defined below is what is used for this in the rest of the tests. Can you reuse/rename/rewrite that function instead?